### PR TITLE
Document support joins and compound selects

### PR DESCRIPTION
**Pull request by carsonfarmer**
Hey @brunocalza can you make sure this is factually correct? I think it is pretty clearcut similar to SQLite, but just make sure I'm not making any wrong assumptions/simplifications?

### CODE REVIEW

File: specs/sql/README.md [hunk 1]
```diff
@@ -896,29 +917,117 @@ right-hand dataset of $N_r$ rows of $M_r$ columns, then the cartesian
 product is a dataset of $N_l \times N_r$ rows, each
```
Comment (brunocalza) [inline specs/sql/README.md:917]: I realized this is not being rendered correctly:

Screen Shot 2023-01-05 at 10 44 26

**Comment by brunocalza** [comment]
> Hey @brunocalza can you make sure this is factually correct? I think it is pretty clearcut similar to SQLite, but just make sure I'm not making any wrong assumptions/simplifications?

Looks correct to me! Both features are identical to SQLite

**Review by brunocalza — APPROVED** [review]

File: specs/sql/README.md (hunk 1 above)
Comment (carsonfarmer) [reply specs/sql/README.md:917]: On GitHub?

Comment (brunocalza) [reply specs/sql/README.md:917]: Yeah. See: https://github.com/tablelandnetwork/docs/tree/main/specs/sql#join-clause

Comment (carsonfarmer) [reply specs/sql/README.md]: Ok, no idea why that was the case. But after a bunch of attempts to fix, I have it working. See https://github.com/tablelandnetwork/docs/blob/a6a636bee10d8d3242026ed77f13dd0b412f640b/specs/sql/README.md#details-9. I ended up just re-writing that "by hand" again, and it works. So there must have been some sort of special utf-8 char in there or something messing things up?

**Pull request merged**